app/Http/Controllers/Admin/SettingsController.php

```diff
@@ -109,7 +109,7 @@ public function update(Request $request)
         $settingsClass = new $settings_class();
 
         foreach ($settingsClass->toArray() as $key => $value) {
-            switch (gettype($value)) {
+            switch (gettype($request->input($key))) {
```

Comment:
> could result in unwanted errors due to type mismatches
If i change the switch statement's default type to just throw in the value that was given instead of setting it to null, as well as adding 'NULL' as it's own switch case, then your comment would be invalidated, and we would shift the "unwanted errors" to not be an issue on our side. Effectively moving the blame to "Code your value types in the front end better" towards whoever changes the front-end, as the backend will not reject other types with the change I could propose..